pkg/network/setup/network.go

```diff
@@ -63,7 +70,11 @@ func (v VMNetworkConfigurator) getPhase2NICs(domain *api.Domain, networks []v1.N
 			continue
 		}
 
-		nic, err := newPhase2PodNIC(v.vmi, &networks[i], iface, v.handler, v.cacheCreator, domain)
+		var attachmentType string
+		if v.domainAttachments != nil {
+			attachmentType = v.domainAttachments[iface.Name]
+		}
+		nic, err := newPhase2PodNIC(v.vmi, &networks[i], iface, v.handler, v.cacheCreator, domain, attachmentType)
```

Comment:
Pass what? I don't see Or's comment anymore.

Reply:
Yes, Or's comment is no longer seen :/

Anyway, what I wrote stands.
This code can be dropped:
```go 
        var attachmentType string
	if v.domainAttachments != nil {
			attachmentType = v.domainAttachments[iface.Name]
		}
```

In favor of what I wrote before.
I think you used the same in some other place.

I won't insist though.

Reply:
Since I"m using `WithDomainAttachments` to inject the `domainAttachments` to `domainAttachments`, I fund it weird to assume it is not `nil`.

Reply:
My suggestion https://github.com/kubevirt/kubevirt/pull/10732#discussion_r1403432176
No need to expect both key & value and no need for nil checking, when `domainAttachments` is nil or interface name not exist, it will return the default value, in this case empty string.

Same goes for other places `domainAttachment` map is referred to (virt-launcher converter code).

Reply:
done.